packages/nimble-components/src/table/specs/table-column-width-hld.md

```diff
@@ -150,11 +156,19 @@ At the moment there is no recognized use case for a pixel-width column that woul
 export class MyPixelWidthColumn : TableColumn<...> {
     public constructor() {
         super();
-        this.columnInternals.currentPixelWidth = 100;
+        this.columnInternals.resizingDisabled = true;
+        this.columnInternals.pixelWidth = 100;
+        this.columnInternals.minPixelWidth = 0;
```

Comment:
It seems a little verbose for a non-resizable column to have to set the `minPixelWidth` in addition to giving a `pixelWidth` and disabling resize, but I was hesitant to make too many assumptions in other parts of the code (particularly the `TableLayoutManager`) about what was true in a non-resizable column.

For example, should the `TableLayoutManager` assume that a non-resizable column has a `pixelWidth` set and is not a fractional width column? While that probably isn't an unreasonable assumption right now since we are only planning on having one non-resizable column and it will be a pixel-width column, I didn't like baking in assumptions that were not enforced anywhere.

Therefore, I decided that having a narrow fixed pixel column update the `minPixelWidth` was better than making those assumptions in the code. However, I'm open to any other suggestions anyone has.

Reply:
Ultimately, I think all the `TableLayoutManager` should care about is whether it needs to resolve a column drag operation to a new `currentFractionalWidth` value or not. So, as I see it, the new `resizingDisabled` state is all that is needed to control appropriate divider visibility behavior for any given column. The state that controls how we [resolve `currentFractionalWidth` after a drag operation](https://github.com/ni/nimble/blob/937a555e4fd5d4928d64a5fb706d083acdc42aad/packages/nimble-components/src/table/models/table-layout-manager.ts#L245) is that column's `pixelWidth` state. If `pixelWidth` is not undefined, we do nothing, otherwise we do what is currently being done.

If that is an accurate understanding of things, then I would think we _don't_ need to require a `minPixelWidth` being set for a fixed width column. I easily could be missing something, and I'm happy to talk through it if needed.

Reply:
After some discussion, we've landed on having the column update the `minPixelWidth` to an appropriate value if it supports being narrower than the default minimum width. Therefore, in this example, the `minPixelWidth` would be set to the same value as the actual width since the column isn't resizable.

Some of the rationale for this is:
- It keeps us from making assumptions in the `TableLayoutManager`, such as when we coerce `currentPixelWidth` to `minPixelWidth` if the `currentPixelWidth` is too small
- Some odd things could happen if the current width of the column is smaller than its minimum width
    - A resizable column in that scenario could get bigger but never be shrunk back to its original size
    - There's a number of places in the `TableLayoutManager` that assume today the `currentPixelWidth` will not be smaller than the column's `minPixelWidth`
- It doesn't seem unreasonable for a column that is explicitly trying to set its size or allow its size to be smaller than the default minimum width to have to set `minPixelWidth`.